### DISCUSSION TITLE
Sync VSCode-web preferences accross workspaces

### DISCUSSION THREAD

**Discussion by artichoked1** [coder/coder#10165] [Feature Request]
Not sure if this can be done or not, but it would be really nice if all the vscode configuration files (like `settings.json`, `extensions.json`, etc.) for templates with code-server built in (`docker`, `docker-dotfiles` ,etc) were synced across all workspace for that user. This would be really handy, because currently, there isn't really a good way to sync settings for each workspace (native sync is disabled and pretty much all sync plugins don't work).

**Comment by Mrc527** [comment]
That’s a lovely idea, I’m also struggling with this issue!

**Comment by artichoked1** [comment]
is there still any possibility of a feature like this happening? Or any other method of syncing settings that could be used in the mean time?

**Comment by matifali** [comment]
Coder provides excellent flexibility for the templates by leveraging any infrastructure and the [level of persistence](https://coder.com/docs/v2/latest/templates/resource-persistence#resource-persistence) for workspaces. You can achieve this by persisting and sharing the location of these files across workspaces.
For example, if appropriate, the same home volume can be mounted to all the workspaces of a developer.

**Comment by matifali** [comment]
This could be solved by https://github.com/coder/modules/pull/411
